agent/agent.go

```diff
@@ -130,6 +131,55 @@ func (a *Agent) Run() []error {
 	return errs
 }
 
+// DryRun runs the agent to log what would occur during a run, without issuing any commands or writing to disk.
+func (a *Agent) DryRun() []error {
+	var errs []error
+
+	a.l.Info("Starting dry run")
+
+	// glob "*" here is to support copy/paste of seeker identifiers
+	// from -dryrun output into select/exclude filters
+	a.tmpDir = "*"
+	a.l.Info("Would copy included files", "includes", a.Config.Includes)
+
+	// Running healthchecks for products. We don't want to stop if any fail though.
+	a.l.Info("Checking product availability")
+	if errProductChecks := a.CheckAvailable(); errProductChecks != nil {
+		errs = append(errs, errProductChecks)
+		a.l.Error("Product failed healthcheck. ensure setup steps are complete", "error", errProductChecks)
+	}
+
+	// Create products and their seekers
+	a.l.Info("Gathering seekers for each product")
+	p, errProductSetup := a.Setup()
+	if errProductSetup != nil {
+		errs = append(errs, errProductSetup)
+		a.l.Error("Failed running Products", "error", errProductSetup)
```

Comment:
Small suggestion on wording here - I could see someone getting a little confused about anything "running" (or failing to do so) when doing a dryrun. 
```suggestion
		a.l.Error("Failed gathering seekers for products", "error", errProductSetup)
```